Arc-enabled SQL Server scenario updates (#197)
Azure/arc_jumpstart_docs: docs/azure_arc_jumpstart/azure_arc_sqlsrv/azure/azure_arm_template_sqlsrv_winsrv_defender/_index.md

  ![Screenshot showing Azure portal deployment of Jumpstart scenario](./portaldeploy.png)

>   > **Note:** Use SQL Server Edition parameter to deploy SQL Server Standard or Enterprise edition on the Hyper-V guest VM. Supported values are Developer, Standard, and Enterprise. Default is Developer edition. Azure Arc-enabled SQL Server features such as performance metrics requires Standard or Enterprise edition.

Author: ```suggestion
  > **Note:** Use SQL Server Edition parameter to deploy SQL Server Standard or Enterprise edition on the Hyper-V guest VM. Supported values are Developer, Standard, and Enterprise. The default is the Developer edition. Azure Arc-enabled SQL Server features such as performance metrics require Standard or Enterprise edition.
```

  ![Screenshot showing Defender for SQL security incidents and alerts](./defender-sql-security-incidents.png)

- The below screenshot shows an email alert sent by Defender for Cloud when a SQL threat is detected. By default, this email is sent to the registered contact email at the subscription level.
  ![Screenshot showing test script results](./brute-force-attack-alert.png)

> > **Note:** Occasionally Microsoft Defender for SQL may not detect SQL threats generated by the simulation script file *testDefenderForSQL.ps1* executed during the Client VM logon and will not receive an alert email as documented in this scenario. If this happens follow steps under **Simulate Microsoft Defender for SQL threats** below to re-generate threats and receive an alert email.

Reviewer: ```suggestion
> **Note:** Occasionally, Microsoft Defender for SQL server may not detect SQL threats generated by the simulation *testDefenderForSQL.ps1* PowerShell script executed during the Client VM logon automation and will not receive an alert email as documented in this scenario. If this happens follow the steps in the **Simulate Microsoft Defender for SQL threats** below to re-generate threats and receive an alert email.
```

## Simulate Microsoft Defender for SQL threats

Follow the steps below to simulate Microsoft Defender for SQL threats

- On the *JS-Client* Azure VM desktop, locate Hyper-V Manager icon as shown below and double click to open Hyper-V Manager.
  ![Screenshot showing Hyper-V Manager desktop icon](./locate-hyper-v-manager.png)

- In the Hyper-V Manager expand JS-Client and double click on the JS-Win-SQL-01 guest VM to login.
  ![Screenshot showing SQL server guest VM in Hyper-V Manager](./locate-guest-sql-vm.png)

- Click connect. Following are the credentials to login to nested SQL Server VM.

  ```text
  Username: Administrator
  Password: JS123!!
  ```

>   ![Screenshot showing to connect SQL server guest VM](./connect-guest-sql-vm.png)

Reviewer: ```suggestion
  ![Screenshot showing how to connect SQL server guest VM](./connect-guest-sql-vm.png)
```

> - Enter Administrator password as *JS123!!* and login to the guest VM.
>   ![Screenshot showing enter password to login to guest VM](./login-to-guest-vm.png)
> 
> - Open Windows Explorer and navigate to C:\Jumpstart\agentScript folder.
>   ![Screenshot showing locating Defender for SQL simulation script file](./locate-simulation-script.png)
> 
> - Right click *testDefenderForSQL.ps1* script file and select *Run with PowerShell*.
>   ![Screenshot showing executing Defender for SQL simulation script file](./execute-defender-simulation-script.png)

Reviewer: ```suggestion
- Enter the Administrator password as *JS123!!* and login to the guest VM.
  ![Screenshot showing enter password to login to guest VM](./login-to-guest-vm.png)

- Open Windows Explorer and navigate to the *C:\Jumpstart\agentScript* folder.
  ![Screenshot showing locating Defender for SQL simulation script file](./locate-simulation-script.png)

- Right click on the *testDefenderForSQL.ps1* PowerShell script file and select *Run with PowerShell*.
  ![Screenshot showing executing Defender for SQL simulation script file](./execute-defender-simulation-script.png)
```

  ```text
  Username: Administrator
  Password: JS123!!
  ```

  ![Screenshot showing manual execution of the test scripts](./manual-brute-force-test.png)

- The below screenshot shows an email alert sent by Defender for Cloud when a SQL threat is detected. By default, this email is sent to the registered contact email at the subscription level.
  ![Screenshot showing test script results](./brute-force-attack-alert.png)

## Basic troubleshooting

> Occassionally Defender for Cloud for SQL may not detect SQL threats generated by the simulation script file _testDefenderForSQL.ps1_ executed during the Client VM logon and will not receive alert email as documented in this scenario. Retry executing this script on the Hyper-V guest VM _JS-Win-SQL-01_ to re-generate threats and receive alert email. Follow the steps below to re-run the _testDefenderForSQL.ps1_ script.

Author: ```suggestion
Occasionally Defender for Cloud for SQL may not detect SQL threats generated by the simulation script file _testDefenderForSQL.ps1_ executed during the Client VM logon and will not receive an alert email as documented in this scenario. Retry executing this script on the Hyper-V guest VM _JS-Win-SQL-01_ to re-generate threats and receive an alert email. Follow the steps below to re-run the _testDefenderForSQL.ps1_ script.
```
